modules/restoring-red-hat-quay-standalone.adoc

```diff
@@ -0,0 +1,214 @@
+:_content-type: PROCEDURE
+[[restoring-red-hat-quay-standalone]]
+= Restoring {productname} on standalone deployments 
+
+This procedure describes how to restore {productname} on standalone deployments. 
+
+.Prerequisites 
+
+* You have backed up your {productname} deployment. 
+
+.Procedure 
+
+. Create a new directory that will bind-mount to `/conf/stack` inside of the {productname} container:
++
+[source,terminal]
+----
+$ mkdir /opt/new-quay-install
+----
+
+. Copy the contents of your temporary backup directory created in xref:backing-up-red-hat-quay-standalone[Backing up {productname} on standalone deployments] to the `new-quay-install1` directory created in Step 1:
++
+[source,terminal]
+----
+$ cp /tmp/quay-backup/quay-backup.tar.gz /opt/new-quay-install/
+----
+
+. Change into the `new-quay-install` directory by entering the following command:
++
+[source,terminal]
+----
+$ cd /opt/new-quay-install/
+----
+
+. Extract the contents of your {productname} directory: 
++
+[source,terminal]
+----
+$ tar xvf /tmp/quay-backup/quay-backup.tar.gz *
+----
++
+Example output:
++
+----
+config.yaml
+config.yaml.bak
+extra_ca_certs/
+extra_ca_certs/ca.crt
+ssl.cert
+ssl.key
+----
+
+. Recall the `DB_URI` from your backed-up `config.yaml` file by entering the following command: 
++
+[source,terminal]
+----
+$ grep DB_URI config.yaml
+----
++
+Example output:
++
+[source,yaml]
+----
+postgresql://<username>:test123@172.24.10.50/quay
+----
+
+. Run the following command to enter the PostgreSQL database server: 
++
+[source,terminal]
+----
+$ sudo postgres 
+----
+
+. Enter psql by running the following command:
++
+[source,terminal]
+----
+$ psql 
+----
+
+. Create a new database by running the following command:
++
+[source,terminal]
+----
+postgres=# "example-restore-registry-quay-database" OWNER "example-restore-registry-quay-database";
+----
++
+Example output:
++
+----
+CREATE DATABASE
+----
+
+. Connect to the database by running the following command:
++
+[source,terminal]
+----
+postgres=# \c "example-restore-registry-quay-database";
+----
++
+Example output:
++
+[source,terminal]
+----
+You are now connected to database "example-restore-registry-quay-database" as user "postgres".
+----
+
+. Create a `pg_trmg` extension of your Quay database by running the following command:
++
+[source,terminal]
+----
+example-restore-registry-quay-database=# create extension pg_trgm ;
+----
++
+Example output:
++
+[source,terminal]
+----
+CREATE EXTENSION
+----
+
+. Exit the postgres CLI by entering the following command: 
++
+[source,terminal]
+----
+\q
+----
+
+. Set the password for your PostgreSQL deployment by running the following command:
++
+[source,terminal]
+----
+$ psql -h 172.24.10.50 -U <username> -W -d quay < /tmp/quay-backup/quay-backup.sql
+----
++
+Example output:
++
+----
+SET
+SET
+SET
+SET
+SET
+----
```

Comment:
Depending on the guide of step 5 to step 12, I suppose this example is to back up one database to another database in the same postgresql server.  So in my verification test,  I was trying to back up data from database `quay` to new database `example_restore_registry_quay_database ` in the same postgresql server `whupsql1.postgres.database.azure.com`.

*  back up `quay` database
```
$ pg_dump -h whupsql1.postgres.database.azure.com -p 5432 -d quay -U w...y -W -O  > /tmp/quay-backup/quay-backup.sql
```
 
* create new database `example_restore_registry_quay_database`
```
$ psql "host=whupsql1.postgres.database.azure.com port=5432 dbname=postgres user=w..y password=A...2"
postgres=> CREATE DATABASE example_restore_registry_quay_database;
CREATE DATABASE

postgres=> \c example_restore_registry_quay_database;

example_restore_registry_quay_database=> CREATE EXTENSION IF NOT EXISTS pg_trgm;
CREATE EXTENSION

example_restore_registry_quay_database=> \q

$ psql "host=whupsql1.postgres.database.azure.com port=5432 dbname=example_restore_registry_quay_database user=w..y " -W <  /tmp/quay-backup/quay-backup.sql
```

Reply:
If customer try to move database from one postgrepsql server to another postgrepsql server.  Customer have to update `DB_URI ` setting in `config.yaml` before restarting the quay.
```
DB_URI: postgresql://<username>:<password>@<new_postgrepsql_host>:5432/example_restore_registry_quay_database

```